cli/deployment/config.go

```diff
@@ -215,9 +215,9 @@ func newConfig() *codersdk.DeploymentConfig {
 				Flag:   "oidc-client-secret",
 				Secret: true,
 			},
-			EmailDomain: &codersdk.DeploymentConfigField[string]{
-				Name:  "OIDC Email Domain",
-				Usage: "Email domain that clients logging in with OIDC must match.",
+			EmailDomains: &codersdk.DeploymentConfigField[[]string]{
+				Name:  "OIDC Email Domains",
+				Usage: "Email domains that clients logging in with OIDC must match.",
 				Flag:  "oidc-email-domain",
```

Comment:
@bpmct what do you think, should we change this to `oidc-email-domains`? A backwards incompatible change, but users will immediately receive an error.

Reply:
I'm fairly sure this change already breaks the environment variable, so we'll need to update our docs in [docs/auth.md](https://github.com/coder/coder/blob/main/docs/admin/auth.md#openid-connect-with-google). 

> users will immediately receive an error

I'm fine doing a breaking change here, but don't entirely understand who/what/where the error displays for. Are all domains suddenly allowed and an error appears in the console? Or is the server unable to start? This seems dangerous.

Reply:
Perspective from user using this environment variable and wanting this. We'd ideally like to keep it the same so all domains aren't suddenly silently allowed. But I'd add the new variable beforehand knowing it's coming. Not a huge issue since we restrict users at the IdP but we do document this as a security feature at our end. In reality, "email" is not really meant to be used for security so it's just a nice to have.

Reply:
Since the config name was changed here, so will the environtment variable be. I had a PR for allowing the env name to be overridden, but we decided not to do it (yet, https://github.com/coder/coder/pull/4894). If we want to maintain the previous env var we should make the config non-plural. But ultimately we should definitely keep these two in sync. Either we break both or none.

@bpmct right now the user is met with zero errors. The env var will silently break and the flag remains as it was. We should consider erroring out when unknown `CODER_` variables are set. But this is problematic since some env vars may be valid for other CLI commands. So perhaps we should maintain a list of deprecated env vars for which we show an error.

Reply:
Fair enough. I have made it all plural.

Reply:
> Either we break both or none.

@dcarrion87 hate for you to undo your work, but let's keep this flag singular for the time being. Would rather be grammatically incorrect over potentially introducing security hole for another user who may not read the changelog 😅 

In the future, we can explore how we can alias flags and/or communicate this to users when upgrading via a banner.

Reply:
@bpmct @mafredri no issues at all, I get it! 😄 

At the risk of sounding stupid what exactly am I changing. Is it just the `oidc-email-domain` or am I changing all the variables back to `Email domain` and `EmailDomain` as well for consistency?

Reply:
You can keep the one in `coderd` plural (since it's internal), but the one in `DeploymentFlags` and the CLI flag (`oidc-email-domain`) should be reverted, thanks!